### DISCUSSION TITLE
Any tips or advice for migrating a project from React Static to Next.js?

### DISCUSSION THREAD

**Discussion by chrisESparkles** [react-static/react-static#1670] [Q&A] [answered]
As per the title, with the discontinuation of React Static, is there helpful advice that the community (and perhaps @SleeplessByte) could give to someone who wants to migrate a React Static application to Next.js?

Particularly, I'd like to know the best place to start, if there are any React Static-specific pain-points to look out for, and if there are equivalents among RS and Next components. Also, if there is some non-RS related recommendations for Next.js you' like to share, that would also be appreciated.

**Comment by SleeplessByte** [comment, marked as answer]
Hi @chrisESparkles , 

The [Next.js Getting Started Docs](https://nextjs.org/docs/getting-started) are pretty decent. A few things to check:

- Routing works the same as in it [uses directory structure](https://nextjs.org/docs/basic-features/pages), but understand it only works for the `pages` folder, and not an arbitrary one (by default).
- [Dynamic routes must also exist as files](https://nextjs.org/docs/basic-features/pages#pages-with-dynamic-routes), whereas in React-Static they don't.
- [Everything is pre-rendered](https://nextjs.org/docs/basic-features/pages#two-forms-of-pre-rendering) instead of excluding pre-rendering on certain routes. This is because _every_ request is always rendered by a server, either at request time or at build time, whereas in React-Static you could only pre-render at build time.
- You must (in most cases) use the [Next.js image component](https://nextjs.org/docs/basic-features/image-optimization) instead of whatever you're using right now. 
- Static files are auto exposed [from the public folder](https://nextjs.org/docs/basic-features/static-file-serving)
- The [build command](https://nextjs.org/docs/deployment#nextjs-build-api) is pretty much the same, but check out the docs anyway.

Good luck!

**Reply by chrisESparkles** [reply]
Thank you 😊
